Top collection data volume and % change

daiagi opened the pull request:
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #4264 
- [x] Closes #4266 
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [x] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
![image](https://user-images.githubusercontent.com/22791238/202167350-d52efbcc-4835-4a38-ab15-6b1348b9bdc4.png)

roiLeo reviewed (changes requested):
Hey!

Why did you remove 3 columns on large screen?

Rmrk chain is not working anymore (same for snek rococo)
![Screenshot 2022-11-15 at 12-15-48 KodaDot - Kusama NFT Market Explorer](https://user-images.githubusercontent.com/9987732/201906468-da3d5294-16c3-49a7-8893-71bae2fb98df.png)

coingecko api is called more than once
![Capture d’écran 2022-11-15 à 12 18 42 PM](https://user-images.githubusercontent.com/9987732/201906970-d98c6813-3aba-4808-912c-73cdf37a7504.png)

something wrong with %change text size:
**Figma**
![Capture d’écran 2022-11-15 à 12 21 10 PM](https://user-images.githubusercontent.com/9987732/201907584-2f0e0289-f8d2-4009-b64c-1aa474ddab69.png)


**Preview**
![Screenshot 2022-11-15 at 12-21-19 KodaDot - Kusama NFT Market Explorer](https://user-images.githubusercontent.com/9987732/201907604-08ebcef5-59c3-4c62-b176-3ff43362a99f.png)

daiagi commented:
1. Restored 3 colums - it needs some css work
2. they work - but as i mentions in #4264 the volume is 0 for 1 month or less time frames 
(I think i'm getting the volume correctly)
   for now i've changed the timeframe default value to "All" 
4. fixed congecko API calls  - it's 1 call now. Thanks 
5. adjusted %percent style 
![image](https://user-images.githubusercontent.com/22791238/201919061-b502de74-b161-446e-ba5e-0f0b0f779f67.png)

roiLeo commented on components/landing/TopCollectionsItem.vue:
```diff
@@ -12,45 +12,127 @@
-}>()
+  get color() {
+    if (this.diffPercent.value) {
+      return this.diffPercent.value < 0 ? 'red' : 'green'
```
```suggestion
      return this.diffPercent.value < 0 ? 'has-text-danger' : 'has-text-success'
```

roiLeo commented on components/landing/TopCollectionsItem.vue:
```diff
@@ -12,45 +12,127 @@
+}
 </script>
+<style scoped>
+.green {
+  color: rgb(0, 188, 0);
+}
+
+.red {
+  color: red;
+}
```
```suggestion
```

roiLeo commented on components/landing/TopCollectionsItem.vue:
```diff
@@ -12,45 +12,127 @@
 </template>
 
-<script lang="ts" setup>
+<script lang="ts">
```
Why did you rewrite it nuxt2 way? 
we try to migrate slowly to nuxt3/vue3 (ref #2690)

daiagi replied:
Honestly... I'm new to Vue , coming from react
I needed access to the store and could not find any example in the code base where it is done NOT using this.$store
so i figured it needs to be a class component

please enlighten me on how to do that properly

roiLeo replied:
You can do something like `const { $store } = useNuxtApp()` to acces Nuxt store

roiLeo commented on components/landing/TopCollections.vue:
```diff
@@ -77,10 +89,11 @@ export default class TopCollections extends mixins(AuthMixin, PrefixMixin) {
 
   async fetch() {
     await this.fetchCollectionsSeries()
+    this.$store.dispatch('fiat/fetchFiatPrice')
```
if getters empty then dispatch action

roiLeo reviewed (commented):
IMO, we should write a resolver for this kind of data (% change)

roiLeo commented on components/landing/TopCollectionsItem.vue:
```diff
@@ -12,24 +12,40 @@
+              <BasicMoney :value="usdValue" inline :unit="'USD'" />
+            </div>
+          </div>
+          <div
+            class="is-flex is-justify-content-center is-align-items-center pl-2 is-size-5">
+            <p :class="color">
+              {{ diffPercentString }}
+            </p>
+          </div>
```
font-size; hide div if empty, not a paragraph
```suggestion
          <div
            v-if="diffPercentString"
            class="is-flex is-justify-content-center is-align-items-center pl-2 is-size-6" :class="color">
            {{ diffPercentString }}
          </div>
```

roiLeo commented on queries/rmrk/subsquid/collectionsSales.graphql:
```diff
@@ -0,0 +1,16 @@
+query collectionsSales($ids: [String!]) {
```
since this query is not used only by rmrk I would suggest changing the directory

daiagi replied:
* Added the range buttons
* yep I'm aware that RMRK chain data is still empty. to my understanding I'm querying the correct data so it is what it is (empty)
please have a look at the query and the way i process it  and let me know if you think I'm displaying wrong data

* although not in the requirements, i've added an 'All' time filter button so you could see that under that filter
 RMRK chain data is not empty

* moved the query to @/queries/collectionsSales.graphql

* point to consider:
before my code changes, the volume in top collections and the volume in collection details page did not match
now they do
so either:
1. I fixed a bug
2. I'm calculating wrong volume in top collection, could be the reason why it's empty

Please advice

roiLeo reviewed (commented):
I don't know what % change I'm looking at but rmrk chain data is still empty

could you show somewhere the range? or show the different options (link #4266)

daiagi commented:
* Added the range buttons
* yep I'm aware that RMRK chain data is still empty. to my understanding I'm querying the correct data so it is what it is (empty)
please have a look at the query and the way i process it  and let me know if you think I'm displaying wrong data

* although not in the requirements, i've added an 'All' time filter button so you could see that under that filter
 RMRK chain data is not empty

* moved the query to @/queries/collectionsSales.graphql

* point to consider:
before my code changes, the volume in top collections and the volume in collection details page did not match
now they do
so either:
1. I fixed a bug
2. I'm calculating wrong volume in top collection, could be the reason why it's empty

Please advice

daiagi commented:
@roiLeo 
have a look, this collection for example indeed didn't have activity in the last month

![image](https://user-images.githubusercontent.com/22791238/202155879-e2c35dd0-59a4-4294-8ecd-e14edaf71045.png)

roiLeo reviewed (approved):
✅ code looks good

// TODO:
- missing translation 
- missing some types `(nft)`
- transform `<TopCollections />` component nuxt3 way
- refactor `timeFilters` with `<NeoButton />` from #4330 
<details>
  <summary>Code</summary>
  
```vue
      <b-field>
        <p class="control">
          <NeoButton :class="{ active: state.timeRange == 'Day' }" label="24 Hours" @click="setTimeRange('Day')" />
        </p>
        <p class="control">
          <NeoButton :class="{ active: state.timeRange == 'Day' }" label="7 Days" @click="setTimeRange('Week')" />
        </p>
        <p class="control">
          <NeoButton :class="{ active: state.timeRange == 'Day' }" label="30 Days" @click="setTimeRange('Month')" />
        </p>
        <p class="control">
          <NeoButton :class="{ active: state.timeRange == 'Day' }" label="All" @click="setTimeRange('All')" />
        </p>
      </b-field>
```
  
</details>

daiagi commented:
@roiLeo 

are these TODOS needed to be done before merging?
should i remove the 'All' time filter?
the PR needs another reviewer, but i can't request one

daiagi commented:
@roiLeo @yangwao 
Can we move this forward?

yangwao commented:
> are these TODOS needed to be done before merging?

- #4364

yangwao commented:
hey looks good to me, 
small changes could be made in follow up issues

- design of time filter for black version, you can find them here https://www.figma.com/file/3iOjW12ERFRDSVnpLuhQmf/landing-page-handoff?node-id=798%3A3662

<img width="518" alt="image" src="https://user-images.githubusercontent.com/5887929/203122499-3a33ba66-69e6-4995-8fb9-3d27c4a8dcc0.png">

- we've meanwhile changed time frames as there is less data since design file, and we went for 7d/30d/90d filters

yangwao commented:
hey I'm curious, hover state seems lagging when circling with cursor around?

is there something special about some running queries on background? 
I don't see anything special in network tab tbh.

https://user-images.githubusercontent.com/5887929/203124941-8ce87118-a0ca-497c-8977-283e00201d65.mov

current beta at `main` 

https://user-images.githubusercontent.com/5887929/203125219-fc034051-5e2e-4eb4-8478-7396c63c3e5f.mov

yangwao commented:
pay 150 usd 
extra lifting! :D 

feel free to continue with
- #4366 
- #4365 
- #4364

yangwao commented:
😍 Perfect, I’ve sent the payout 
💵 $150 @ 23.7 USD/KSM ~ 6.329 $KSM
🧗 [EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L](https://kusama.subscan.io/account/EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)
🔗 [0x2e3234f7aac9cfbecf238162acf40d1ea95700390d8186db6c45731fadf62e7e](https://kusama.subscan.io/extrinsic/0x2e3234f7aac9cfbecf238162acf40d1ea95700390d8186db6c45731fadf62e7e)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

roiLeo commented:
Where 3 columns? Did you remove the breakpoint?

> hey I'm curious, hover state seems lagging when circling with cursor around?
> 
> is there something special about some running queries on background? I don't see anything special in network tab tbh.

wfm
